SimTKcommon/Scalar/include/SimTKcommon/internal/NTraits.h

```diff
@@ -799,9 +799,14 @@ template <class R> class NTraits< conjugate<R> > {
     static TNormalize normalize(const T& t) {return TNormalize(t/abs(t));}
 
     // 1/conj(z) = conj(1/z), for complex z.
-    static TInvert invert(const T& t)    
-    {   const typename NTraits<THerm>::TInvert cmplx(NTraits<THerm>::invert(t.conj()));
-        return reinterpret_cast<const TInvert&>(cmplx); } // recast complex to conjugate it
+    // Here reinterpret_cast is source of [-Wstrict-aliasing] warning.
+    // New code is more explicit and should be faster(???).
+    static TInvert invert(const T& t){
+      return static_cast<TInvert>(TReal(1)/t.conj());
```

Comment:
Thank you for explanation! I'm aware about project approach to just reinterpret underlying values for performance sake. As you mentioned conjugate and complex need to be _same type_ to fix this warning, I believe this warning along with optimizer could introduce strange bugs hard to trace.